torch_geometric/nn/models/g_retriever.py

```diff
@@ -0,0 +1,248 @@
+        gnn_in_channels (int): (default: :obj:`1024`)
+        gnn_hidden_channels (int): (default: obj:`1024`)
+        gnn_out_channels (int): (default: :obj:`1024`)
+        num_gnn_layers (int): (default: :obj:`4`)
+        num_gnn_heads (int): Number of heads to use for BasicGNNs with the
+            :obj:`heads` kwarg. (default: :obj:`4`)
+        mlp_hidden_dim (int): (default: :obj:`2048`)
+        mlp_out_dim (int): (default: :obj:`4096`)
+
+    .. warning::
+        This module has been tested with the following Hugging Face models
+
+        - :obj:`llm_to_use="meta-llama/Llama-2-7b-chat-hf"`
+        - :obj:`llm_to_use="google/gemma-7b"`
+
+        and may not work with other models. See other models at `Hugging Face
+        Models <https://huggingface.co/models>`_ and let us know if you
+        encounter any issue by submitting a GitHub issue.
+
+    .. note::
+        See `examples/llm_plus_gnn/g_retriever.py` for example usage.
+    """
+    def __init__(
+        self,
+        llm_to_use='meta-llama/Llama-2-7b-chat-hf',
+        llm_use_lora: bool = False,
+        llm_dtype=torch.bfloat16,
+        num_llm_params: int = 7,
+        gnn_to_use=GAT,
+        gnn_in_channels: int = 1024,
+        gnn_hidden_channels: int = 1024,
+        gnn_out_channels: int = 1024,
+        num_gnn_layers: int = 4,
+        num_gnn_heads: int = 4,
+        mlp_hidden_dim: int = 2048,
+        mlp_out_dim: int = 4096,
+    ) -> None:
+        super().__init__()
+        self.llm_to_use = LLM(llm_to_use, num_llm_params, llm_dtype)
+        self.llm_generator = self.llm_to_use.llm
+        self.llm_dtype = llm_dtype
+        if llm_use_lora:
+            from peft import (
+                LoraConfig,
+                get_peft_model,
+                prepare_model_for_kbit_training,
+            )
+            self.llm_generator = prepare_model_for_kbit_training(
+                self.llm_generator)
+            lora_r: int = 8
+            lora_alpha: int = 16
+            lora_dropout: float = 0.05
+            lora_target_modules = [
+                "q_proj",
+                "v_proj",
+            ]
+            config = LoraConfig(
+                r=lora_r,
+                lora_alpha=lora_alpha,
+                target_modules=lora_target_modules,
+                lora_dropout=lora_dropout,
+                bias="none",
+                task_type="CAUSAL_LM",
+            )
+            self.llm_generator = get_peft_model(self.llm_generator, config)
+        self.llm_device = self.llm_to_use.llm_device
+        self.tokenizer = self.llm_to_use.tokenizer
+        try:
+            self.graph_encoder = gnn_to_use(
+                in_channels=gnn_in_channels,
+                out_channels=gnn_out_channels,
+                hidden_channels=gnn_hidden_channels,
+                num_layers=num_gnn_layers,
+                heads=num_gnn_heads,
+                norm='batch_norm',
+            ).to(self.llm_device)
+        except:  # noqa
```

Comment:
I'm not in favor of simply adding `noqa`, but at least it should have a specific rule code to ignore, e.g., E501 for too long line length.